src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs

```diff
@@ -291,7 +309,8 @@ public double Reduction
                     }
                 }
 
-                public double TopKAccuracy { get { return _numInstances > 0 ? _numCorrectTopK / _numInstances : 0; } }
+                public double TopKAccuracy => !(OutputTopKAcc is null) ? AllTopKAccuracy[OutputTopKAcc.Value-1] : 0d;
+                public double[] AllTopKAccuracy => CumulativeSum(_seenRanks.Take(OutputTopKAcc ?? 0).Select(l => l / (double)(_numInstances - _numUnknownClassInstances))).ToArray();
```

Comment:
Should we be subtracting away `_numUnknownClassInstances`? The previous `TopKAccuracy` simply divided by `_numInstances` ([code](https://github.com/dotnet/machinelearning/blob/610ffcb67083c2e5e6e1a14884ba24b1da0384c7/src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs#L294)).

If there are new unseen labels in the scoring dataset, I would expect the last value in the TopKAccuracies array not be `1.0`, since there is no rank at which I will receive the right answer. The right answer can not be predicted by the model as the label was not seen in the training dataset.

Example when all scoring datasets labels are known:
`{ 0.1, 0.3, 0.7, 0.9, 1.0 }` (ends with `1.0`)

Example scoring datasets includes unseen labels:
`{ 0.1, 0.3, 0.7, 0.9, 0.95 }` (doesn't end with `1.0`)

Reply:
[Here](https://github.com/dotnet/machinelearning/issues/744#issuecomment-433327217) it was stated that:

> // Note: All unseen classes are not part of the accuracy (similar to how other implementations handles this).
// So we substract the unseen classes. This makes sense since in the end we will train the model on ALL data and this case will be unlikely.

I don't know what "other implementations" he was referring to. But I think you're right in that the previous implementation didn't take the difference with the unknown class instances, so I think I can change that.

Reply:
> ...so I think I can change that.

Sounds good. It's better to be self-consistent anyhow; the other metrics merely divide by `_numInstances` ([1](https://github.com/dotnet/machinelearning/blob/610ffcb67083c2e5e6e1a14884ba24b1da0384c7/src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs#L254), [2](https://github.com/dotnet/machinelearning/blob/610ffcb67083c2e5e6e1a14884ba24b1da0384c7/src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs#L276)).

Reply:
Updated, and added a test for this